src/main/java/com/metamx/emitter/service/CloudServiceEmitter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2012 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.metamx.emitter.service;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.metamx.emitter.core.Emitter;
+
+public class CloudServiceEmitter extends ServiceEmitter
```

Comment:
I don't really see how the enforcement works. If you just update the emitter library, nobody enforces you to replace ServiceEmitter usages with CloudServiceEmitter. So adding another constructor to ServiceEmitter has the same effect. Nullable fields are also not a problem, IMO. getServer() and getHost() are already nullable.

I don't like the idea of adding a new class because it pollutes the API. When we will decide to add another fields, should we create another one class? Should it extend ServiceEmitter directly, or CloudServiceEmitter? What if we want some of fields of being not defined on purpose?